### ISSUE TITLE
Publish a resolved tensor-in/tensor-out dense encoding kit

### ISSUE THREAD

**Issue by clemsgrs** [clemsgrs/slide2vec#267]
## Summary

Publish a resolved, tensor-in/tensor-out dense encoding module for consumers that encode **inside a training or inference loop after augmentation**.

The existing persistence interfaces are intentionally not a fit: `embed_images_dense` and `embed_regions_dense` read sources and write artifacts, while a live consumer already has augmented pixels, needs an on-device grid immediately, and discards that grid after the step.

The public seam should be a cohesive `DenseEncodeKit`, not five unrelated runtime exports.

## Ownership seam

- The consumer owns paired image/mask reading, joint augmentation, task training, and prediction.
- After augmentation, the consumer hands slide2vec one RGB `uint8` CHW tensor.
- Slide2vec owns everything that determines the encoder input and dense grid: shipped normalization, geometry validation, bottom/right padding, device transfer, frozen encoder construction, evaluation mode, `no_grad`, autocast, whole/sliding encoding, attention selection, precision, and output dtype.
- No source reading, cache lookup, persistence, or artifact creation occurs on this path.

## Public interface

Illustrative shape:

```python
kit = model.prepare_dense_encoder(
    dense=DenseImageOptions(...) | DenseOptions(...),
    execution=ExecutionOptions(...),
)

preprocess = kit.preprocessor()
item = preprocess(augmented_rgb_uint8_chw)
grids = kit.encode(collated_cpu_batch)
```

### `Model.prepare_dense_encoder(...)`

- Accept either existing dense option type and resolve their shared encoding fields through one implementation.
- Source-reading fields such as spacing, backend, and tolerance have no effect after the augmented-pixel handoff and must be documented as outside this interface.
- Perform check-before-load validation from registry metadata wherever possible: target geometry, padding mode, window geometry, feature kind, attention settings, and variable-input capability fail before loading the multi-gigabyte encoder.
- Load/configure the frozen encoder through the same input-contract and precision path as persisted dense extraction, then assert runtime metadata matches the preflight plan.
- Return one shareable kit; consumers may reuse it across folds.

### `DenseEncodeKit`

- Is a plain public object, deliberately **not** an `nn.Module`.
- Keeps the foundation encoder private so consumers cannot accidentally register it in a decoder, optimizer, or checkpoint.
- Keeps the encoder frozen and in evaluation mode.
- Exposes only:
  - immutable `geometry`;
  - `preprocessor()`;
  - `encode(batch)`.

### Worker-safe preprocessor

- `preprocessor()` returns a lightweight serializable callable carrying only the shipped normalization transform and resolved geometry.
- Accept exactly one RGB `torch.Tensor` with shape `(3, H, W)` and dtype `uint8`.
- Apply the shipped normalization transform itemwise, validate the declared target geometry, and pad bottom/right to the patch multiple.
- Return one CPU `(3, Henc, Wenc)` tensor for normal DataLoader collation.
- Reject batched, non-CHW, non-RGB, or wrong-dtype input with actionable errors.
- Batching starts after preprocessing; `encode` is the batched interface.

### Batched encoder

- `encode(batch)` accepts a collated CPU `(B, 3, Henc, Wenc)` tensor.
- Own device transfer, using non-blocking transfer when applicable.
- Run under evaluation mode, `torch.no_grad()`, and slide2vec's resolved autocast policy.
- Honor whole-image and sliding-window modes.
- Honor both `patch_features` and `cls_attention`, including attention blocks and register-token selection.
- Return an on-device `(B, D, Gh, Gw)` tensor in the same resolved output dtype as persisted dense extraction.
- The returned grid has no gradient history.

### Public geometry

`kit.geometry` is the authoritative immutable live geometry and exposes:

- `target_size`;
- `patch_size`;
- `encoded_size`;
- `grid_shape`;
- bottom/right `pad`;
- top-left-anchored `crop_box`.

A consumer must not independently recompute encoder geometry for the live path.

## Implementation direction

The existing internal `DenseGridEncoder` is the implementation substrate: it already centralizes normalization, geometry, padding, feature selection, and sliding encode. Do not expose it unchanged—its current `encode_batch()` detaches to CPU NumPy and leaves autocast outside, which does not satisfy the live on-device tensor contract.

Do not make `pad_image_to_encoded`, `encode_dense_sliding`, `resolve_window_geometry`, or `slide_encode_autocast_ctx` individually public. Their ordering and invariants belong behind the kit interface.

The generic `cover_origins` helper used by Soma's outer prediction canvas is task-level inference behavior and is not part of this interface.

## Acceptance criteria

- [ ] Focused public-interface tests are written first.
- [ ] `Model.prepare_dense_encoder(...)`, `DenseEncodeKit`, its preprocessor, and its geometry type are available from documented public imports.
- [ ] Invalid geometry/options fail before encoder loading whenever registry metadata is sufficient.
- [ ] The worker preprocessor is serializable and applies the exact normalization/validation/padding path used by persisted dense extraction.
- [ ] `encode` accepts a CPU batch, owns device transfer/frozen eval/`no_grad`/autocast, and returns an on-device tensor in the resolved output dtype.
- [ ] A deterministic literal encoder proves exact `torch.equal` parity with persisted extraction for all four modes:
  - whole-image + `patch_features`;
  - sliding-window + `patch_features`;
  - whole-image + `cls_attention`;
  - sliding-window + `cls_attention`.
- [ ] The preprocessor output is exactly equal to the persisted extraction preprocessing output for explicit example inputs.
- [ ] The encoder is not registered through the kit, remains frozen/eval, and output tensors have no gradient history.
- [ ] Calling the kit creates no files, cache entries, sidecars, or embedding artifacts.
- [ ] Shape, channel, dtype, and geometry failures identify the violated contract.
- [ ] Public API and release documentation explain the augmented-pixel handoff and tensor shapes.
- [ ] Existing focused and full tests pass.

## Downstream adoption

- clemsgrs/soma#322 adopts the released kit and removes dense production imports from `slide2vec.runtime.*`.
- clemsgrs/soma#288 separately removes Soma's non-dense runtime imports through the already-public `Model.embed_tiles` interface.

## Out of scope

- Foundation-encoder gradients or fine-tuning.
- Image/slide/mask reading.
- Joint image/mask augmentation.
- Cache policy, persistence, sidecars, or artifacts.
- Soma adoption in this repository issue.
- Soma live slide-manifest ROI support.
- Outer prediction-window stitching or a public `cover_origins` utility.
